ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/AEMDownloadManger.java

```diff
@@ -147,24 +147,47 @@ private void enqueueExtractAemImportsFromManifests() {
         }
     }
 
+    @AnyThread
+    public ThreadPoolExecutor enqueueSyncManifestAemImports(Manifest manifest, Boolean force) {
```

Comment:
it should be returning a ListenableFuture, you will create the ListenableFuture that you are returning in the `SyncAemImportTask` constructor. Then you will complete it in the `run` method of the `SyncAemImportTask`

Reply:
You shouldn't need to modify any of the executors or use any (aside from the MainThreadExecutor) at all to complete this